pkg/antctl/raw/traceflow/command.go

```diff
@@ -0,0 +1,337 @@
+	}
+
+	tf, err := newTraceflow()
+	if err != nil {
+		return fmt.Errorf("error when filling up traceflow config: %w", err)
+	}
+
+	if _, err = client.OpsV1alpha1().Traceflows().Create(context.TODO(), tf, v1.CreateOptions{}); err != nil {
+		return fmt.Errorf("error when creating traceflow, is traceflow feature gate enabled? %w", err)
+	}
+	defer func() {
+		if waiting {
+			if err = client.OpsV1alpha1().Traceflows().Delete(context.TODO(), tf.Name, v1.DeleteOptions{}); err != nil {
+				klog.Errorf("error when deleting traceflow: %w", err)
+			}
+		}
+	}()
+
+	if !waiting {
+		return nil
+	}
+
+	if err := wait.Poll(1*time.Second, 15*time.Second, func() (bool, error) {
+		tf, err := client.OpsV1alpha1().Traceflows().Get(context.TODO(), tf.Name, v1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if tf.Status.Phase != v1alpha1.Succeeded {
+			return false, nil
+		}
+		output(tf)
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("error when retrieving traceflow: %w", err)
+	}
+
+	return nil
+}
+
+func setupKubeconfig(kubeconfig *rest.Config, groupVersion *schema.GroupVersion) {
+	kubeconfig.APIPath = "/apis"
+	kubeconfig.GroupVersion = groupVersion
+	kubeconfig.NegotiatedSerializer = scheme.Codecs.WithoutConversion()
+	kubeconfig.Insecure = true
+	kubeconfig.CAFile = ""
+	kubeconfig.CAData = nil
+}
+
+func newTraceflow() (*v1alpha1.Traceflow, error) {
+	var name string
+	var src v1alpha1.Source
+	split := strings.Split(source, "/")
+	if len(split) == 1 {
+		src.Namespace = "default"
+		src.Pod = split[0]
+	} else if len(split) == 2 && len(split[0]) != 0 && len(split[1]) != 0 {
+		src.Namespace = split[0]
+		src.Pod = split[1]
+	} else {
+		return nil, fmt.Errorf("source should be in the format of namespace/pod or pod")
+	}
+
+	var dst v1alpha1.Destination
+	dstIP := net.ParseIP(destination)
+	if dstIP != nil {
+		dst.IP = dstIP.String()
+		name = randName(fmt.Sprintf(fmt.Sprintf("%s-%s-to-%s-", src.Namespace, src.Pod, dst.IP)))
+	} else {
+		split = strings.Split(destination, "/")
+		if len(split) == 1 {
+			dst.Namespace = "default"
+			dst.Pod = split[0]
+		} else if len(split) == 2 && len(split[0]) != 0 && len(split[1]) != 0 {
+			dst.Namespace = split[0]
+			dst.Pod = split[1]
+		} else {
+			return nil, fmt.Errorf("destination should be in the format of namespace/pod or pod")
+		}
+		name = randName(fmt.Sprintf("%s-%s-to-%s-%s-", src.Namespace, src.Pod, dst.Namespace, dst.Pod))
+	}
+
+	pkt, err := parseFlow()
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse flow: %w", err)
+	}
+
+	tf := &v1alpha1.Traceflow{
+		ObjectMeta: v1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1alpha1.TraceflowSpec{
+			Source:      src,
+			Destination: dst,
+			Packet:      *pkt,
+		},
+	}
+
+	return tf, nil
+}
+
+func parseFlow() (*v1alpha1.Packet, error) {
+	flow = strings.ReplaceAll(flow, " ", "")
+	pkt := new(v1alpha1.Packet)
+	protocols := map[string]int32{
+		"icmp": 1,
+		"tcp":  6,
+		"udp":  17,
```

Comment:
We should organize these magic numbers in future.

Reply:
Add const value for these.